### PR TITLE
Migrate every components files to TypeScript

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | It's a part of the typescript refacto
| Type?             | refacto
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | yes
| How to test?      | Some BO pages should be explored, specially those using component files
| Possible impacts? | Whole BO JS

⚠️ Wait for https://github.com/PrestaShop/PrestaShop/pull/24613 to be merged

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24654)

### CODE REVIEW

File: admin-dev/themes/new-theme/js/components/choice-table.ts [hunk 1]
```diff
@@ -33,20 +33,27 @@ export default class ChoiceTable {
    * Init constructor
    */
   constructor() {
-    $(document).on('change', '.js-choice-table-select-all', (e) => {
-      this.handleSelectAll(e);
-    });
+    $(document).on(
+      'change',
+      '.js-choice-table-select-all',
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/choice-table.ts]: Could you move to selector map ?

File: admin-dev/themes/new-theme/js/components/form-submit-button.ts [hunk 2]
```diff
@@ -53,50 +53,59 @@ const {$} = window;
  */
 export default class FormSubmitButton {
   constructor() {
-    $(document).on('click', '.js-form-submit-btn', function (event) {
-      event.preventDefault();
+    $(document).on(
+      'click',
+      '.js-form-submit-btn',
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/form-submit-button.ts]: Could you move it to Selector Map ?

File: admin-dev/themes/new-theme/js/components/grid/extension/action/row/category/delete-category-row-action-extension.js [hunk 3]
```diff
@@ -1,72 +0,0 @@
...
- *
- * @author    PrestaShop SA and Contributors <contact@prestashop.com>
- * @copyright Since 2007 PrestaShop SA and Contributors
- * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
- */
-
-const {$} = window;
-
-/**
- * Class CategoryDeleteRowActionExtension handles submitting of row action
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/grid/extension/action/row/category/delete-category-row-action-extension.js:29]: Really deleted ?

**Comment by NeOMakinG** [comment]
@Progi1984 This PR contains like 30 files from the PR adding typescript, it's too hard to review for now you already reviewed a lot inside it

File: admin-dev/themes/new-theme/js/components/grid/extension/column/catalog/category-position-extension.ts [hunk 4]
```diff
@@ -64,22 +72,32 @@ export default class CategoryPositionExtension {
...
+  handleCategoryPositionChange(row: HTMLElement): void {
     const positions = decodeURIComponent($.tableDnD.serialize());
-    const way = (this.originalPositions.indexOf(row.id) < positions.indexOf(row.id)) ? 1 : 0;
+    const way = this.originalPositions.indexOf(row.id) < positions.indexOf(row.id)
+      ? 1
+      : 0;
 
-    const $categoryPositionContainer = $(row).find(`.js-${this.grid.getId()}-position:first`);
+    const $categoryPositionContainer = $(row).find(
+      `.js-${this.grid.getId()}-position:first`,
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/grid/extension/column/catalog/category-position-extension.ts]: Please move a selector map

File: admin-dev/themes/new-theme/js/components/grid/extension/submit-grid-action-extension.ts [hunk 5]
```diff
@@ -50,19 +52,25 @@ export default class SubmitGridActionExtension {
...
     }
 
-    const $form = $(`#${grid.getId()}_filter_form`);
+    const $form = $(GridMap.filterForm(grid.getId()));
 
     $form.attr('action', $submitBtn.data('url'));
     $form.attr('method', $submitBtn.data('method'));
-    $form.find(`input[name="${grid.getId()}[_token]"]`).val($submitBtn.data('csrf'));
+    $form
+      .find(`input[name="${grid.getId()}[_token]"]`)
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/grid/extension/submit-grid-action-extension.ts]: Move it to selector map

File: admin-dev/themes/new-theme/js/components/module-card.ts [hunk 6]
```diff
@@ -75,13 +107,17 @@ export default class ModuleCard {
...
-  initActionButtons() {
+  initActionButtons(): void {
     const self = this;
 
     $(document).on('click', this.forceDeletionOption, function () {
       const btn = $(
         self.moduleActionModalUninstallLinkSelector,
-        $(`div.module-item-list[data-tech-name='${$(this).attr('data-tech-name')}']`),
+        $(
+          `div.module-item-list[data-tech-name='${$(this).attr(
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/module-card.ts]: Please move to selector map

File: admin-dev/themes/new-theme/js/components/module-card.ts [hunk 7]
```diff
@@ -91,57 +127,81 @@ export default class ModuleCard {
...
-    $(document).on('click', this.moduleActionMenuInstallLinkSelector, function () {
-      if ($('#modal-prestatrust').length) {
-        $('#modal-prestatrust').modal('hide');
-      }
+    $(document).on(
+      'click',
+      this.moduleActionMenuInstallLinkSelector,
+      function () {
+        if ($('#modal-prestatrust').length) {
+          $('#modal-prestatrust').modal('hide');
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/module-card.ts]: Please move to selector map

File: admin-dev/themes/new-theme/js/components/module-card.ts [hunk 8]
```diff
@@ -194,50 +261,74 @@ export default class ModuleCard {
...
+        return self.requestToController(
+          'disable',
           $(
-            self.moduleActionMenuUninstallLinkSelector,
-            $(`div.module-item-list[data-tech-name='${$(e.target).attr('data-tech-name')}']`),
+            self.moduleActionMenuDisableLinkSelector,
+            $(
+              `div.module-item-list[data-tech-name='${$(this).attr(
+                'data-tech-name',
+              )}']`,
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/module-card.ts]: Please move to selector map (and it's the same as beyond)

File: admin-dev/themes/new-theme/js/components/module-card.ts [hunk 9]
```diff
@@ -194,50 +261,74 @@ export default class ModuleCard {
...
+        $(e.target)
+          .parents('.modal')
+          .on('hidden.bs.modal', () => self.requestToController(
+            'uninstall',
+            $(
+              self.moduleActionMenuUninstallLinkSelector,
+              $(
+                `div.module-item-list[data-tech-name='${$(e.target).attr(
+                  'data-tech-name',
+                )}']`,
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/module-card.ts]: :top:

File: admin-dev/themes/new-theme/js/components/multi-store-restriction-field/multi-store-restriction-field.ts [hunk 10]
```diff
@@ -87,9 +94,14 @@ export default class MultiStoreRestrictionField {
...
    */
-  toggleSourceFieldByTargetElement($targetElement, isDisabled) {
+  private toggleSourceFieldByTargetElement(
+    $targetElement: JQuery,
+    isDisabled: boolean,
+  ): void {
     const targetValue = $targetElement.data('shopRestrictionTarget');
-    const $sourceFieldSelector = $(`[data-shop-restriction-source="${targetValue}"]`);
+    const $sourceFieldSelector = $(
+      `[data-shop-restriction-source="${targetValue}"]`,
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/multi-store-restriction-field/multi-store-restriction-field.ts]: Plesae move to selector map

File: admin-dev/themes/new-theme/js/components/multiple-choice-table.ts [hunk 11]
```diff
@@ -33,15 +33,19 @@ export default class MultipleChoiceTable {
    * Init constructor
    */
   constructor() {
-    $(document).on('click', '.js-multiple-choice-table-select-column', (e) => this.handleSelectColumn(e));
+    $(document).on(
+      'click',
+      '.js-multiple-choice-table-select-column',
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/multiple-choice-table.ts]: Please move to selector map (duplicate again)

File: admin-dev/themes/new-theme/js/components/multiple-choice-table.ts [hunk 12]
```diff
@@ -51,7 +55,11 @@ export default class MultipleChoiceTable {
     const $table = $selectColumnBtn.closest('table');
 
     $table
-      .find(`tbody tr td:nth-child(${$selectColumnBtn.data('column-num')}) input[type=checkbox]`)
+      .find(
+        `tbody tr td:nth-child(${$selectColumnBtn.data(
+          'column-num',
+        )}) input[type=checkbox]`,
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/multiple-choice-table.ts]: Selector Map

File: admin-dev/themes/new-theme/js/components/translatable-field.ts [hunk 13]
```diff
@@ -33,40 +33,40 @@ const {$} = window;
...
+  translationFieldSelector: string;
+
+  selectedLocale: string;
+
+  constructor(options: Record<string, any>) {
     const opts = options || {};
 
-    this.localeButtonSelector = opts.localeButtonSelector || '.translationsLocales.nav .nav-item a[data-toggle="tab"]';
+    this.localeButtonSelector = opts.localeButtonSelector
+      || '.translationsLocales.nav .nav-item a[data-toggle="tab"]';
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/translatable-field.ts]: Selector Map

File: admin-dev/themes/new-theme/js/components/translatable-field.ts [hunk 14]
```diff
@@ -94,14 +94,17 @@ class TranslatableField {
...
+  toggleFields(event: Record<string, string>): void {
     this.selectedLocale = event.selectedLocale;
+
     $(this.localeNavigationSelector).each((index, navigation) => {
       const selectedLink = $('.nav-item a.active', navigation);
       const selectedLocale = selectedLink.data('locale');
 
       if (this.selectedLocale !== selectedLocale) {
-        $(`.nav-item a[data-locale="${this.selectedLocale}"]`, navigation).tab('show');
+        $(`.nav-item a[data-locale="${this.selectedLocale}"]`, navigation).tab(
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/components/translatable-field.ts]: Selector Map

**Review by Progi1984 — COMMENTED** [review]
Some selector maps and it's approved

File: admin-dev/themes/new-theme/js/components/grid/extension/action/row/category/delete-category-row-action-extension.js (hunk 3 above)
Comment (NeOMakinG) [reply admin-dev/themes/new-theme/js/components/grid/extension/action/row/category/delete-category-row-action-extension.js:29]: Mooved to TS file

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by Progi1984** [comment]
@NeOMakinG Please rebase your PR

**Comment by NeOMakinG** [comment]
Can be closed, these changes are in another PR, this one is too hard to rebase

**Pull request closed without merging**